src/rgw/rgw_rest_s3.cc

```diff
@@ -5870,3 +5875,321 @@ bool rgw::auth::s3::S3AnonymousEngine::is_applicable(
 
   return route == AwsRoute::QUERY_STRING && version == AwsVersion::UNKNOWN;
 }
+
+
+using namespace s3selectEngine;
+const char* RGWSelectObj_ObjStore_S3::header_name_str[3] = {":event-type", ":content-type", ":message-type"};
+const char* RGWSelectObj_ObjStore_S3::header_value_str[3] = {"Records", "application/octet-stream", "event"};
+
+RGWSelectObj_ObjStore_S3::RGWSelectObj_ObjStore_S3():
+  s3select_syntax(std::unique_ptr<s3selectEngine::s3select>(new s3select())),
+  m_s3_csv_object(std::unique_ptr<s3selectEngine::csv_object>(((s3selectEngine::csv_object*)0))),
```

Comment:
this is the same thing as default-constructing the std::unique_ptr. you can just omit this line entirely along with `crc32` below